torch/utils/data/dataloader.py

```diff
@@ -564,6 +564,8 @@ def _get_shared_seed(self):
             _shared_tensor_seed = torch.empty((), dtype=torch.int64).random_(generator=self.generator)
             if dist.is_available() and dist.is_initialized():
                 rank = dist.get_rank()
+                if dist.get_backend() == "nccl":
+                    _shared_tensor_seed = _shared_tensor_seed.cuda(rank)
```

Comment:
This is not the best way to solve the problem, because for multi-machine case the rank doesn't represent the `device_id` in the process.